src/ray/raylet/raylet.cc

```diff
@@ -74,7 +74,12 @@ Raylet::Raylet(instrumented_io_context &main_service,
                     node_name,
                     node_manager_config,
                     object_manager_config,
-                    gcs_client_),
+                    gcs_client_,
+                    /*set_agent_info_and_register_node*/
+                    [this](const AgentInfo &agent_info) {
+                      self_node_info_.mutable_agent_info()->CopyFrom(agent_info);
+                      RAY_CHECK_OK(RegisterGcs());
+                    }),
```

Comment:
It looks bad. Why do it this way? We should try to decouple node_manager_ with the Raylet.

To be clear, I'm not comfortable that NodeManager <-> Raylet has circular dependence here.

Reply:
How about move the logic of `RegisterGcs ` from `Raylet` to `NodeManager`? Maybe registering node should be a part of `NodeManager`?